framework/platform/bpl/uci/cfg/bpl_cfg.cpp

```diff
@@ -114,9 +114,9 @@ int cfg_get_stop_on_failure_attempts()
 {
     int retVal = -1;
     if (cfg_get_prplmesh_param_int("stop_on_failure_attempts", &retVal) == RETURN_ERR) {
-        MAPF_ERR("cfg_get_stop_on_failure_attempts: Failed to read stop_on_failure_attempts "
+        MAPF_INFO("cfg_get_stop_on_failure_attempts: Failed to read stop_on_failure_attempts "
                  "parameter\n");
-        return RETURN_ERR;
+        return DEFAULT_STOP_ON_FAILURES_ATTEMPTS;
```

Comment:
This is still not the way I would like it.

I would prefer to add a second version of `cfg_get_prplmesh_param_int` like this:

```
int cfg_get_prplmesh_param_int(const std::string &param, int *buf, int default)
{
    int status;
    char val[MAX_UCI_BUF_LEN] = "";

    status = cfg_get_prplmesh_param(param, val, MAX_UCI_BUF_LEN);
    if (status == RETURN_ERR) {
        // Not found, use default
        *buf = default;
        return RETURN_OK;
    }
    status = sscanf_s(val, "%d", buf);
    if (status != 1)
        return RETURN_ERR;

    return RETURN_OK;
}
```

and use that version here.

Reply:
OK this makes sense I guess cause it will reduce code so I'm not against it as I previousely was.
I prefer the following signature:
```
int cfg_get_prplmesh_param_int(const std::string &param, int *value, int default)
```

Then it is used as follows:
```
int cfg_get_rdkb_extensions()
{
    int value;
    if (cfg_get_prplmesh_param_int("rdkb_extensions", &value, DEFAULT_RDKB_EXTENSIONS) == RETURN_OK) {
        return value;
    }
    return RETURN_ERR;
}
```

And we should do the same for other parameter types - it will save a lot of duplicated code.
I guess this was the purpose for this change to begin with, so my appologies @vitalii-komisarenko  for changing my mind again :)